pkg/estimator/client/interface.go

```diff
@@ -30,26 +30,38 @@ import (
 const UnauthenticReplica = -1
 
 var (
-	replicaEstimators              = map[string]ReplicaEstimator{}
-	unschedulableReplicaEstimators = map[string]UnschedulableReplicaEstimator{}
+	replicaEstimators              []ReplicaEstimator
+	unschedulableReplicaEstimators []UnschedulableReplicaEstimator
 )
 
 // ReplicaEstimator is an estimator which estimates the maximum replicas that can be applied to the target cluster.
 type ReplicaEstimator interface {
 	MaxAvailableReplicas(ctx context.Context, clusters []*clusterv1alpha1.Cluster, replicaRequirements *workv1alpha2.ReplicaRequirements) ([]workv1alpha2.TargetCluster, error)
+	GetPriority() EstimatorPriority
```

Comment:
In my opinion, a estimator should not declare priority for `itself`, the priority is reflected by consumers how to use it.

Reply:
Maybe we could decide the default priority? I think accurate scheduler estimator should not be enabled along with other estimators.

Reply:
Yeah, I agree. the accurate scheduler estimator can bypass the general estimator by default.

But there is a corner case we need to consider:
What if some of the clusters do not enable the `karmada-estimator`?  Or, what if some of the `karmada-estimator`s can not estimate the replicas at that time(such as timeout)? Should the scheduler back off to `general estimator`?

Reply:
Got it, this is possible. Fall back off to general estimator could be a nice choice.

Reply:
I have a new perspective here, but I haven't had time to update and implement it:

if the higher-priority estimators have gaven results for all members clusters, we don't need to call lower-priority estimators.

if the higher-priority estimators only gave part of result for members clusters, which means some of the member clusters didn't get result, we then call lower-priority estimators, but its result only apply to member clusters which didn't get result.

I mean those member clusters which already been successfully calculated by `karmada-estimator`, no need back off to `general`, back off only aims at those not be successfully calculated.

Reply:
Yes, for the cluster which already estimated by its `karmada-estimator` no need to back off to `general estimator`.

Reply:
done